crates/blockchain/payload.rs

```diff
@@ -442,18 +486,14 @@ impl TransactionQueue {
                     sender: tx.sender,
                 };
                 // Insert head into heads list while maintaing order
-                let mut index = 0;
-                loop {
-                    if self
-                        .heads
-                        .get(index)
-                        .is_some_and(|current_head| compare_heads(current_head, &head).is_gt())
-                    {
-                        index += 1;
-                    } else {
-                        self.heads.insert(index, head.clone());
-                    }
-                }
+                let index = match self
+                    .heads
+                    .binary_search_by(|current_head| compare_heads(current_head, &head))
```

Comment:
Are we 100% sure that heads is ordered? What happens if its not?

Reply:
The heads are sorted upon creation of the TransactionQueue and every operation maintains this order, it is the sole purpose of the struct